src/libertem/io/dataset/raw_csr.py

```diff
@@ -230,6 +230,11 @@ def supports_correction(self):
     @classmethod
     def detect_params(cls, path: str, executor: "JobExecutor"):
         try:
+            _, extension = os.path.splitext(path)
+            has_extension = extension.lstrip('.') in cls.get_supported_extensions()
+            under_size_lim = os.stat(path).st_size < 2**20  # 1 MB
```

Comment:
For the remote use-case, need to do fs access via the executor:

```suggestion
            under_size_lim = executor.run_function(os.stat, path).st_size < 2**20  # 1 MB
```

Reply:
Good spot, thanks.